sunpy/io/jp2.py

```diff
@@ -72,5 +74,107 @@ def get_header(filepath):
     return [FileHeader(pydict)]
 
 
+def header_to_xml(header):
+    """
+    Converts image header metadata into an XML Tree that can be inserted into
+    a JP2 file header.
+
+    Parameters
+    ----------
+    header : `MetaDict`
+        A header dictionary to convert to xml.
+
+    Returns
+    ----------
+    `lxml.etree._Element`
+        A fits element where each child is an xml element
+        in the form <key>value</key> derived from the key/value
+        pairs in the given header dictionary
+    """
+    # glymur uses lxml and will crash if trying to use
+    # python's builtin xml.etree
+    import lxml.etree as ET
+
+    fits = ET.Element("fits")
+
+    already_added = set()
+    for key in header:
+        # Some headers span multiple lines and get duplicated as keys
+        # header.get will appropriately return all data, so if we see
+        # a key again, we can assume it was already added to the xml tree.
+        if (key in already_added):
+            continue
+
+        # Add to the set so we don't duplicate entries
+        already_added.add(key)
+
+        el = ET.SubElement(fits, key)
+        data = header.get(key)
+        if type(data) == bool:
+            data = "1" if data else "0"
+        else:
+            data = str(data)
+
+        el.text = data
+
+    return fits
+
+
+def generate_jp2_xmlbox(header):
+    """
+    Generates the JP2 XML box to be inserted into the jp2 file.
+
+    Parameters
+    ----------
+    header : `MetaDict`
+        A header dictionary.
+
+    Returns
+    ----------
+    `XMLBox`
+        XML box containing FITS metadata to be used in jp2 headers
+    """
+    # glymur uses lxml and will crash if trying to use
+    # python's builtin xml.etree
+    import lxml.etree as ET
+    from glymur import jp2box
+
+    header_xml = header_to_xml(header)
+    meta = ET.Element("meta")
+    meta.append(header_xml)
+    tree = ET.ElementTree(meta)
+    return jp2box.XMLBox(xml=tree)
+
+
 def write(fname, data, header):
```

Comment:
Can you check if other writes have a ``**kwargs``? If so, we should try to follow that for the time being.